clang/lib/Sema/Sema.cpp

```diff
@@ -1753,7 +1753,11 @@ void Sema::checkDeviceDecl(const ValueDecl *D, SourceLocation Loc) {
     if (Ty->isDependentType())
       return;
 
-    if ((Ty->isFloat16Type() && !Context.getTargetInfo().hasFloat16Type()) ||
+    if ((Ty->isFloat16Type() && !Context.getTargetInfo().hasFloat16Type() &&
+         // Disable check for SYCL CUDA BE until FP16 support is properly
+         // reported there (issue#1799)
+         !StringRef("nvptx64-nvidia-cuda-sycldevice")
+              .equals(Context.getTargetInfo().getTriple().str())) ||
```

Comment:
I would do this like here
https://github.com/intel/llvm/blob/f9b6b5e44f7d40ef1a98be2f1492f3b2c77a7496/clang/lib/Sema/SemaType.cpp#L1524

Reply:
thank you for the example. it looks much better.